compiler/rustc_data_structures/src/sorted_map.rs

```diff
@@ -215,32 +216,36 @@ impl<K: Ord, V> SortedMap<K, V> {
     /// It is up to the caller to make sure that the elements are sorted by key
     /// and that there are no duplicates.
     #[inline]
-    pub fn insert_presorted(&mut self, elements: Vec<(K, V)>) {
-        if elements.is_empty() {
+    pub fn insert_presorted(
+        &mut self,
+        // We require `TrustedLen` to ensure that the `splice` below is actually efficient.
+        mut elements: impl Iterator<Item = (K, V)> + DoubleEndedIterator + TrustedLen,
```

Comment:
could use `IntoIterator<Item = (K, V), Iter: DoubleEndedIterator + TrustedLen>` to avoid needing to change callers, but since it's mostly tests, doesn't really matter